### DISCUSSION TITLE
Can't change user password with ReactJs and NodeJS as backend in Supabase

### DISCUSSION THREAD

**Discussion by Alucard2169** [supabase/supabase#17223] [Questions]
I am trying to change user's password with NodeJS as backend in supabase 

This is the hook reponsible for sending request to backend

` 
 import { useState } from "react";

const usePasswordReset =  () => {
    const [data, setData] = useState(null)
    const [error, setError] = useState(null)
    const [isLoading, setIsLoading] = useState(false)
    
    const resetPassword = async ({ email, password }: { email: String, password: String }) => {
        try {
            setIsLoading(true)
            const response = await fetch("/api/resetPassword", {
                method: "POST",
                  headers: {
            "Content-Type": "application/json",
          },body:JSON.stringify({email,password})
            })

            const data = await response.json();
            console.log(data)
            if (!response.ok) {
                setData(null)
                setIsLoading(false)
                throw Error(data.error)
            }
            setIsLoading(false)
            setError(null)
            setData(data)
        }
        catch (error: any) {
            setData(null)
            setIsLoading(false)
            setError(error.message)
        }
    }

    return { data, error, isLoading, resetPassword };
}

export default usePasswordReset;
`

and this is the controller function responsible for perform the actual task

`const resetPassword = async (req:Request,res:Response) => {
  try {
    const { email, password }: { email: string, password: string } = req.body;

    const { data, error } = await supabase.auth.updateUser({
      email,
      password,
    });
    console.log(error);
    if (error) {
      throw new Error(error.message)
    }
    else {
      console.log(data)
      res.json(data)
    }

  } catch (error: any) {
    
  }
}`   

The error i get is (AuthSessionMissingError: Auth session missing!)

I kinda know what is wrong with this, like i don't have a supabase session in my backend, but how do i fix it.

Please any help would be appreciated even if it's a link to the topic in docs in cause i missed it

**Comment by saltcod** [comment]
Hey there! Discord is probably the best place to tackle this one, with lots of questions on approach and setup.

**Reply by Alucard2169** [reply]
I tried posting there 2 times, and waited a week, no response. But I think I will ditch NodeJS and just perform serverless tasks with react and supabase
